engine/source/materials/mat/mat127/sigeps127.F90

```diff
@@ -341,103 +332,106 @@ subroutine sigeps127(                                       &
-                       yc_over_sc  = fourth*(yc(i)/sc(i))**2
-                       emc = fourth*(signzz(i)/sc(i))**2  + (yc_over_sc - one)*signzz(i)/yc(i) + (signxy(i)/sc(i))**2 
-                       if(emc >= one) dmg(i,8)= one 
-                endif 
+               ! matrix failure only on shear 
                 if(abs(signxy(i)) >= sc(i) ) dmg(i,6 ) = one 
-                if(off(i) == four_over_5) then
-                     ndx_fail = ndx_fail + 1
-                     indx_fail(ndx_fail) = i
-                endif
+                ! failur based on effective strain
               end do ! ndex 
-            end select ! twoway     
-        endif ! dfailt == zero
-       ! criteria based on strain
+            end select ! twoway    
+      ! failure criteria 
       ndx_fail = 0
       if (dfailt > zero )then
-#include "vectorize.inc"                        
+          select case (twoway)
+            case(0) ! one fiber direction
+#include "vectorize.inc"              
               do n=1,ndex
                 i= index(n) 
                 ! failure based on max strain 
-                if(dmg(i,5) == one ) then  ! add condition on the matix direction 
-                 xc(i) = ycfac*yc(i)! 
-                 xt(i) = fbrt*xt(i)
-                endif    
                 eps_ef =  two_third* (epsxx(i)**2 + epsyy(i)**2 + epsxy(i)**2 ) 
                 eps_ef = sqrt(eps_ef) 
                 if(epsxx(i) >= dfailt .or. epsxx(i) <= dfailc .or.  abs(epszz(i)) >=dfailm .or.    &
                     abs(epsyy(i)) >= dfailm .or. abs(epsxy(i)) >= dfails  .or. eps_ef >= efs  ) then
-                    dmg(i,1) = one 
+                     dmg(i,1) = one 
                      off(i) = four_over_5
                      ndx_fail = ndx_fail + 1
                      indx_fail(ndx_fail) = i
                      uvar(i,1) = one  
-                else
-                  !  fiber direction checking strengh
-                    if(signxx(i) >= xt(i) ) then
-                      dmg(i,2) = one
-                    elseif(signxx(i) <=  -xc(i)) then
-                      dmg(i,3) = one 
-                    endif 
-                   !  matrix  direction checking strengh direction b 
-                    if(signyy(i) >= yt(i) ) then
-                       dmg(i,4) = one
-                    elseif(signyy(i) <= -yc(i)) then
-                       dmg(i,5) = one 
-                    endif  
-                  !  matrix  direction checking max strengh direction c 
-                    if(signzz(i) >= yt(i) ) then
-                      dmg(i,7) = one
-                    elseif(signzz(i) <= -yc(i)) then
-                      dmg(i,8) = one 
-                    endif  
-                    if(abs(signxy(i)) >= sc(i) ) dmg(i,6 ) = one 
-                endif
-              end do ! nel   
+                 endif  
+              end do   
+            case(1) ! two fiber direction
+#include "vectorize.inc"              
```

Comment:
Vectorize cannot be used here, because of ndx_fail variable update (loops iterations are not independent)